### PR TITLE
Added tests for the "redirect" example

### PR DESCRIPTION
**Pull request by chazelton331**

### CODE REVIEW

File: examples/redirect/tests/redirect.rs [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+            }
+        }
+    });
+}
+
+#[test]
+fn test_root_response_body() {
+    test_get_path!("/", |mut response: Response| {
+        let body_string = response.body().and_then(|body| body.into_string());
+
```
Comment (SergioBenitez) [inline examples/redirect/tests/redirect.rs]: Remove this empty line.

File: examples/redirect/tests/redirect.rs [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+            } else if h.name == "Content-Length" {
+                assert_eq!("0", h.value);
+            }
+        }
+    });
+}
+
+#[test]
+fn test_root_response_body() {
+    test_get_path!("/", |mut response: Response| {
```
Comment (SergioBenitez) [inline examples/redirect/tests/redirect.rs]: Merge this in with the previous test. Just have a single `test_root` function.

File: examples/redirect/tests/redirect.rs [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+        assert_eq!(Status::SeeOther, response.status());
+    });
+}
+
+#[test]
+fn test_root_redirect_headers() {
+    test_get_path!("/", |response: Response| {
+        let headers = response.headers();
+
+        for h in headers {
```
Comment (SergioBenitez) [inline examples/redirect/tests/redirect.rs]: Do `for h in response.headers()` directly.

File: examples/redirect/tests/redirect.rs [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+#[test]
+fn test_root_response_status() {
+    test_get_path!("/", |response: Response| {
+        assert_eq!(Status::SeeOther, response.status());
+    });
+}
+
+#[test]
+fn test_root_redirect_headers() {
+    test_get_path!("/", |response: Response| {
```
Comment (SergioBenitez) [inline examples/redirect/tests/redirect.rs]: Merge the body check in here, too: `assert!(response.body().is_none())`.

File: examples/redirect/tests/redirect.rs [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+
+#[test]
+fn test_login_response_status() {
+    test_get_path!("/login", |response: Response| {
+        assert_eq!(Status::Ok, response.status());
+    });
+}
+
+#[test]
+fn test_login_response_body() {
```
Comment (SergioBenitez) [inline examples/redirect/tests/redirect.rs]: Merge this test with the previous into a single `test_login` function.

File: examples/redirect/tests/redirect.rs [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+
+extern crate rocket;
+extern crate redirect;
+
+use rocket::testing::MockRequest;
+use rocket::Response;
+use rocket::http::Method::*;
+use rocket::http::Status;
+
+macro_rules! test_get_path {
```
Comment (SergioBenitez) [inline examples/redirect/tests/redirect.rs]: Let's just call this `run_test`.

File: examples/redirect/src/main.rs [hunk 7]
```diff
@@ -1,19 +1,9 @@
...
-fn root() -> Redirect {
-    Redirect::to("/login")
-}
 
-#[get("/login")]
-fn login() -> &'static str {
-    "Hi! Please log in before continuing."
-}
+extern crate rocket;
+extern crate redirect;
```
Comment (SergioBenitez) [inline examples/redirect/src/main.rs]: Why has this been split up into a separate crate? There's no ned for that in this example. Simple add a new file `tests.rs`, and add `#[cfg(test)] mod tests` to `src/main.rs`. All of the examples, except for `config`, which is a special case, do this.

**Comment by chazelton331** [comment]
Thanks for the feedback! I think I addressed it all.

File: examples/redirect/src/tests.rs [hunk 8]
```diff
@@ -0,0 +1,38 @@
...
+    })
+}
+
+#[test]
+fn test_root() {
+    run_test!("/", |mut response: Response| {
+        assert!(response.body().is_none());
+        assert_eq!(Status::SeeOther, response.status());
+        for h in response.headers(){
+            if h.name == "Location" {
```
Comment (SergioBenitez) [inline examples/redirect/src/tests.rs]: Let's use a `match` here:

```rust 
match h.name {
    "Location" => assert_eq!(h.value, "/login"),
    "Content-Length" => assert_eq!(h.value.parse(), 0),
    _ => { /* let these through */ }
}
```

**Comment by SergioBenitez** [comment]
Waiting on the last changes. :)

**Comment by chazelton331** [comment]
@SergioBenitez sorry for the delay, should be good now

**Comment by SergioBenitez** [comment]
Awesome! Thank you. :)

File: examples/redirect/src/tests.rs [hunk 9]
```diff
@@ -0,0 +1,38 @@
...
+    })
+}
+
+#[test]
+fn test_root() {
+    run_test!("/", |mut response: Response| {
+        assert!(response.body().is_none());
+        assert_eq!(response.status(), Status::SeeOther);
+        for h in response.headers() {
+            match h.name.as_ref() {
```
Comment (SergioBenitez) [inline examples/redirect/src/tests.rs]: Due to recent changes, this will have to be changed to `h.name.as_str()`. Sorry about that!

**Comment by SergioBenitez** [comment]
Merged in 83e33cf0bef56d46b97815760bf0498c0db24ecf.

**Pull request closed without merging**